telegraf/1.18/alpine/Dockerfile

```diff
@@ -29,6 +29,10 @@ RUN set -ex && \
 
 EXPOSE 8125/udp 8092/udp 8094
 
+RUN addgroup -S telegraf
+RUN adduser -S telegraf -G telegraf
```

Comment:
Have you run the alpine image and edited /etc/telegraf/telegraf.conf? To begin testing we should spot test plugins. We don't have a comprehensive test suite we can run the docker image through

Reply:
@reimda, I created a simple test case:

```
[[inputs.cpu]]
[[outputs.file]]
```

And ran with that config via a bindmount, which correctly launched the CPU plugin:

```
+ docker create --name telegraf --mount type=bind,source=/home/powersj/config.toml,destination=/etc/telegraf/telegraf.conf,readonly test-telegraf
4ce5f054b058d824ad91e548619e228c021cbca33fd3bcccadeea16697288982
+ docker start telegraf
telegraf
+ docker exec telegraf whoami
telegraf
+ docker exec telegraf telegraf --version
Telegraf 1.20.2 (git: HEAD f721f53d)
+ docker exec telegraf telegraf --debug --test --config /etc/telegraf/telegraf.conf
2021-10-18T19:01:48Z I! Starting Telegraf 1.20.2
2021-10-18T19:01:48Z D! [agent] Initializing plugins
2021-10-18T19:01:48Z D! [agent] Starting service inputs
2021-10-18T19:01:49Z D! [agent] Stopping service inputs
2021-10-18T19:01:49Z D! [agent] Input channel closed
2021-10-18T19:01:49Z D! [agent] Stopped Successfully
> cpu,cpu=cpu0,host=4ce5f054b058 usage_guest=0,usage_guest_nice=0,usage_idle=100,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=0,usage_user=0 1634583709000000000
> cpu,cpu=cpu1,host=4ce5f054b058 usage_guest=0,usage_guest_nice=0,usage_idle=100,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=0,usage_user=0 1634583709000000000
> cpu,cpu=cpu2,host=4ce5f054b058 usage_guest=0,usage_guest_nice=0,usage_idle=7.999999999992724,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=7.9999999999998295,usage_user=84.00000000000034 1634583709000000000
> cpu,cpu=cpu3,host=4ce5f054b058 usage_guest=0,usage_guest_nice=0,usage_idle=100,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=0,usage_user=0 1634583709000000000
> cpu,cpu=cpu-total,host=4ce5f054b058 usage_guest=0,usage_guest_nice=0,usage_idle=76.9999999999527,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=2.0000000000003126,usage_user=20.999999999999375 1634583709000000000
```

Is this the kind of test you were thinking? Are there other plugins you think might be good?

Reply:
That's about what I had in mind. I mostly wanted to see that there aren't any easy permissions/ownership problems.

We should probably also test some of the input plugins that are meant to work under docker. Could you check one or two that have a host_proc setting? Those let you bind mount the host's proc directory into the container and gather metrics from it. disk, processes, wireless, bond, and diskio are like this.

The docker input plugin itself has an 'endpoint' option that lets you map the docker host's /var/run/docker.sock into the container so you can gather metrics about the docker host.

I'm sure there are plugins that will be affected by this change, for example ping, snmp_trap, and wireguard all recommend using setcap to give telegraf network privileges. Those will be affected by losing root, but they should act like telegraf does when run as non-root without docker, and that is described in their readme.md files.

There are many other plugins that recommend using sudo or have a use_sudo option, like varnish, lvm, iptables, and smart. Those will also be affected but again they're documented. Many of them also didn't work out of the box before this change because the programs they run have never been included in telegraf's docker image.

Reply:
re: docker endpoint

When the user bind mounts `/var/run/docker.sock` it is going to have root:docker group id permissions. This means running as the Telegraf user by default will fail. The user would need to add something like `--group-add $(stat -c '%g' /var/run/docker.sock)` to their command to get this to work. Is there someone who knows enough if this is acceptable? It seems better than messing with the socket's permissions and is better to be explicit about giving the container permission to mess with the socket. [example with and without group-add](https://paste.ubuntu.com/p/kHQfNXcqf4/)

re: host_proc

With the disk input plugin, this [worked as expected](https://paste.ubuntu.com/p/njkPjhT8zT/) with no additional charges if the host filesystem is bind mounted into the container. If I set the optional env variables it also correctly strips the leading path as well. 

 Same situation with [processes input plugin](https://paste.ubuntu.com/p/tDFdk4FH2B/)

Reply:
re: docker endpoint

How should users who run telegraf as a kubernetes ds use/enable this?